analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,296 @@
+            sourceArtifact = RemoteArtifact(
+                url = vcsHost?.getArchiveDownloadUrl(vcsInfo).orEmpty(),
+                hash = Hash.NONE
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromGithubUserAndProject(id: String, revision: String): Package {
+        val projectUrl = "https://github.com/$id"
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = "$projectUrl.git",
+            revision = revision
+        )
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = VcsHost.GITHUB.getUserOrOrganization(projectUrl).orEmpty(),
+                name = VcsHost.GITHUB.getProject(projectUrl).orEmpty(),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = projectUrl,
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact(
+                url = VcsHost.GITHUB.getArchiveDownloadUrl(vcsInfo).orEmpty(),
+                hash = Hash.NONE
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromLocalRepository(fileUrl: String, revision: String): Package {
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = fileUrl,
+            revision = revision
+        )
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = "",
+                name = fileUrl.substringAfterLast("/"),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = "",
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact.EMPTY,
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromBinarySpec(binarySpec: Map<String, String>, id: String, revision: String) = Package(
+        id = Identifier(
+            type = managerName,
+            namespace = "",
+            name = id.substringAfterLast("/").removeSuffix(".json"),
+            version = revision
+        ),
+        declaredLicenses = sortedSetOf(),
+        description = "",
+        homepageUrl = "",
+
+        // The content from the artifactUrl might contain header files that should be scanned.
+        // Therefore adding this URL for both sourceArtifact and binaryArtifact.
+        binaryArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        sourceArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        vcs = VcsInfo.EMPTY
+    )
+
+    private fun isGithubUserAndProject(id: String): Boolean {
+        val regex = Regex("([a-zA-Z0-9]+)/([a-zA-Z0-9]+)")
```

Comment:
Consider moving this to a private top-level constant / companion object property so the regex does not get recompiled on every function call.

Reply:
Replaced the regex with a simpler `usernameProject.split("/").size == 2` as I might fail to predict how users name their projects.
A `/` in the project name is not allowed and gets converted to `-` by Github.